extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -94,6 +118,37 @@ class AmpYoutube extends AMP.BaseElement {
   }
 
   /** @private */
+  getPlayerReadyPromise_() {
+    return new Promise(resolve => {
+      this.getWin().addEventListener('message', event => {
+        const data = JSON.parse(event.data);
+        if (event.origin != 'https://www.youtube.com' ||
+            event.source != this.iframe_.contentWindow) {
+          return;
+        }
+        if (data.event == 'onReady') {
+          resolve(this.iframe_);
+        } else if (data.event == 'infoDelivery') {
```

Comment:
It's sort of like that. You do:
```
/** @private @const {!Promise} */
this.playerReadyPromise_ = new Promise(resolve => {
  /** @private @const {function()} */
  this.playerReadyResolver_ = resolve;
});

if (data.event == 'onReady') {
  this.playerReadyResolver_();
}
```

We usually try to avoid these, but in this case, I think it's justified.